float8_experimental/fsdp_utils.py

```diff
@@ -57,18 +57,16 @@ def precompute_float8_dynamic_scale_for_fsdp(module: nn.Module) -> None:
 
     # inf-norm is equivalent to max(abs(w))
     max_weights = torch._foreach_norm(weights, ord=math.inf)  # Partial
-    amax_tensor = torch.vstack(max_weights)  # Partial
+    amax_tensor = torch.stack(max_weights)  # Partial
```

Comment:
This is a TLDR for me, I like vstack because semantically I think about gluing lego blocks together lol Does it assert some contiguousnous that causes it to be less performant?

Reply:
From offline: `vstack` incurs per-tensor `reshape`, which each redispatches through `DTensor` dispatch, which is what makes it slow. `stack` only goes through `DTensor` dispatch once.